File: app/src/main/java/com/rabtman/acgclub/di/component/ScheduleDetailComponent.java
package com.rabtman.acgclub.di.component;

import com.rabtman.acgclub.di.module.ScheduleDetailModule;
import com.rabtman.acgclub.mvp.ui.activity.ScheduleDetailActivity;
import com.rabtman.common.di.component.AppComponent;
import com.rabtman.common.di.scope.ActivityScope;
import dagger.Component;

/**
 * @author Rabtman
 */
@ActivityScope
@Component(modules = ScheduleDetailModule.class, dependencies = AppComponent.class)
public interface ScheduleDetailComponent {

  void inject(ScheduleDetailActivity activity);
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/di/component/ScheduleDetailComponent.kt
package com.rabtman.acgschedule.di.component

import com.rabtman.acgschedule.di.module.ScheduleDetailModule
import com.rabtman.acgschedule.mvp.ui.activity.ScheduleDetailActivity
import com.rabtman.common.di.component.AppComponent
import com.rabtman.common.di.scope.ActivityScope
import dagger.Component

/**
 * @author Rabtman
 */
@ActivityScope
@Component(modules = [ScheduleDetailModule::class], dependencies = [AppComponent::class])
interface ScheduleDetailComponent {
    fun inject(activity: ScheduleDetailActivity)
}